### PR TITLE
Injection Range Action - CSE creator

### PR DESCRIPTION
**Pull request by baptsegu**

### CODE REVIEW

File: data/crac-creation/crac-creator-cse/src/main/java/com/farao_community/farao/data/crac_creation/creator/cse/remedial_action/TRemedialActionAdder.java [hunk 1]
```diff
@@ -198,81 +195,85 @@ private void importHvdcRangeAction(TRemedialAction tRemedialAction) {
...
+        GeneratorHelper generatorFromHelper = new GeneratorHelper(hvdcNodes.getFromNode().getV(), ucteNetworkAnalyzer);
+        GeneratorHelper generatorToHelper = new GeneratorHelper(hvdcNodes.getToNode().getV(), ucteNetworkAnalyzer);
 
         // ---- Only handle ABSOLUTE variation type
         if (!tRemedialAction.getHVDCRange().getVariationType().getV().equals(ABSOLUTE_VARIATION_TYPE)) {
             cseCracCreationContext.addRemedialActionCreationContext(
-                CseHvdcCreationContext.notImported(tRemedialAction, hvdcHelper.getUcteId(), ImportStatus.NOT_YET_HANDLED_BY_FARAO, String.format("HVDC %s is not defined with an ABSOLUTE variation type (only ABSOLUTE is handled)", raId))
-            );
+                    CseHvdcCreationContext.notImported(tRemedialAction,
+                            ImportStatus.NOT_YET_HANDLED_BY_FARAO,
```
Comment (phiedw) [inline data/crac-creation/crac-creator-cse/src/main/java/com/farao_community/farao/data/crac_creation/creator/cse/remedial_action/TRemedialActionAdder.java:206]: It looks like your intellij is set to 8 spaces on this project

File: data/crac-creation/crac-creator-cse/src/main/java/com/farao_community/farao/data/crac_creation/creator/cse/remedial_action/TRemedialActionAdder.java [hunk 2]
```diff
@@ -198,81 +195,85 @@ private void importHvdcRangeAction(TRemedialAction tRemedialAction) {
...
-                    cseCracCreationContext.addRemedialActionCreationContext(CseHvdcCreationContext.notImported(tRemedialAction, hvdcHelper.getUcteId(), ImportStatus.OTHER, hvdcHelper.getInvalidReason()));
-                    return;
-                }
-            }
-        }
 
-        // Invert ranges if isInverted
-        double minRange = isInverted ? -tRemedialAction.getHVDCRange().getMax().getV() : tRemedialAction.getHVDCRange().getMin().getV();
-        double maxRange = isInverted ? -tRemedialAction.getHVDCRange().getMin().getV() : tRemedialAction.getHVDCRange().getMax().getV();
+        // ---- check if generator are presents
```
Comment (phiedw) [inline data/crac-creation/crac-creator-cse/src/main/java/com/farao_community/farao/data/crac_creation/creator/cse/remedial_action/TRemedialActionAdder.java:224]: generators* are present*

File: data/crac-creation/crac-creator-cse/src/main/java/com/farao_community/farao/data/crac_creation/creator/cse/remedial_action/TRemedialActionAdder.java [hunk 3]
```diff
@@ -198,81 +195,85 @@ private void importHvdcRangeAction(TRemedialAction tRemedialAction) {
...
+        if (!generatorFromHelper.isValid() || !generatorToHelper.isValid()) {
+            GeneratorHelper invalidGenerator = generatorFromHelper.isValid() ? generatorToHelper : generatorFromHelper;
+            cseCracCreationContext.addRemedialActionCreationContext(
+                    CseHvdcCreationContext.notImported(tRemedialAction,
+                            ImportStatus.ELEMENT_NOT_FOUND_IN_NETWORK,
+                            invalidGenerator.getDetail(),
+                            hvdcNodes.getFromNode().getV(),
+                            hvdcNodes.getToNode().getV()));
+            return;
```
Comment (phiedw) [inline data/crac-creation/crac-creator-cse/src/main/java/com/farao_community/farao/data/crac_creation/creator/cse/remedial_action/TRemedialActionAdder.java:233]: if both are invalid, do we want to add both contexts ?

**Review by phiedw — CHANGES_REQUESTED** [review]

File: data/crac-creation/crac-creator-cse/src/main/java/com/farao_community/farao/data/crac_creation/creator/cse/remedial_action/TRemedialActionAdder.java (hunk 3 above)
Comment (baptsegu) [reply data/crac-creation/crac-creator-cse/src/main/java/com/farao_community/farao/data/crac_creation/creator/cse/remedial_action/TRemedialActionAdder.java:233]: we add one context with the two values of the CRAC (fromNode, toNode), but with any of the associated generators, as they might have not been found in the network.

File: data/crac-creation/crac-creator-cse/src/main/java/com/farao_community/farao/data/crac_creation/creator/cse/remedial_action/TRemedialActionAdder.java (hunk 2 above)
Comment (baptsegu) [reply data/crac-creation/crac-creator-cse/src/main/java/com/farao_community/farao/data/crac_creation/creator/cse/remedial_action/TRemedialActionAdder.java:224]: corrected

**Pull request merged**